### PR TITLE
Add geopotential height diagnostic

### PR DESCRIPTION
**Pull request by tcclevenger**
Previously, `VerticalLayer` diags were computed from sea level (i.e., `surf_geopotential=0`). This adds two diagnostics, `z_*` and `geopotential_*` where `z_*` is calculated with `surf_geopotential=0` and `geopotential_*` with `surf_geopotential=phis(icol)`.

### CODE REVIEW

**Review by AaronDonahue — COMMENTED** [review]
@tcclevenger This is great.  Thanks for putting it together so quickly and I like the elegant solution.  I am happy to approve but I want to get feedback from @PeterCaldwell and/or @golaz on the name of the output which is switching to z_int and z_mid.

**Comment by PeterCaldwell** [comment]
I like the naming convention, but I'd also like Chris's opinion.

**Comment by AaronDonahue** [comment]
@golaz , when you get a chance can you comment on the naming convention in this PR?

**Comment by golaz** [comment]
@tcclevenger , @AaronDonahue  : thanks for implementing this new variable. The naming convention is clear and reasonable.

File: components/eamxx/src/diagnostics/vertical_layer_interface.cpp
Comment (bartgol) [inline components/eamxx/src/diagnostics/vertical_layer_interface.cpp:7]: Do you think it's worth it to also merge the int and mid version into a single `VerticalLayer` diagnostic? There seem to be a lot of code duplication, and the only difference is an extra call to `calculate_z_mid` in the mid case (something that we could handle with a simple if statement)...

Comment (tcclevenger) [reply components/eamxx/src/diagnostics/vertical_layer_interface.cpp:7]: Yeah, a lot of duplicated code. I'll add that.

**Comment by tcclevenger** [comment]
@bartgol Combined all the VerticalLayer* diags. Using `dz` for the old `VerticalLayerThickness`.

File: components/eamxx/src/diagnostics/vertical_layer.cpp
Comment (bartgol) [inline components/eamxx/src/diagnostics/vertical_layer.cpp:120]: The base class already takes care of setting the timestamp in the diag field. so this is not needed.

File: components/eamxx/src/diagnostics/vertical_layer.hpp
Comment (bartgol) [inline components/eamxx/src/diagnostics/vertical_layer.hpp:12]: The class does much more now. We should write in the comment all 5 possible quantities that it can compute.

**Review by bartgol — CHANGES_REQUESTED** [review]
Nice cleanup! There's definitely one thing to fix (string used on device), and a storage/runtime easy optimization in the diag implementation.

**Review by bartgol — DISMISSED** [review]

**Comment by tcclevenger** [comment]
@bartgol accidentally undid some changes with master rebase. I'll need another approval.

**Review by bartgol — DISMISSED** [review]

**Review by bartgol — DISMISSED** [review]

**Comment by tcclevenger** [comment]
@bartgol It looks like https://github.com/E3SM-Project/scream/pull/2327 has added fails to master. Commit d35ea9d1938c66cac6a3a6314091a08236d5cbf7 here fixes the issues with PG2 test, the other is an issue with time_iterpolation test.

**Comment by bartgol** [comment]
@tcclevenger it should have not added fails to master. It was passing everything, except for v1 baselines cmp. But it did change our tests to use ne2 IC/topo files, so yes, your commit should fix that.

**Comment by tcclevenger** [comment]
I saw the same fails for `time_interpolation_np1` in https://github.com/E3SM-Project/scream/pull/2330 which doesn't touch any src code. It looks like it doesn't like the way the PointGrid is setup in the test.

**Comment by AaronDonahue** [comment]
> I saw the same fails for `time_interpolation_np1` in #2330 which doesn't touch any src code. It looks like it doesn't like the way the PointGrid is setup in the test.

I'm working on a fix now.  Should be in this morning.  On a separate note, I also see fails on the CDashboard in the `[homme_shoc_cld_spa_p3_rrtmgp_pg2_np2](https://my.cdash.org/test/81705760)` tests that I think are unrelated.

**Comment by tcclevenger** [comment]
Setting WIP until AT fixes are merged.

**Review by bartgol — APPROVED** [review]

**Pull request merged**
